app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ComponentType.scala

```diff
@@ -89,6 +89,12 @@ object ChoiceType {
   implicit val format: OFormat[ChoiceType] = derived.oformat
 }
 
+case class HmrcTaxPeriod(idType: String, idNumber: String, regimeType: String) extends ComponentType
```

Comment:
Would it be possible to represent `idType`, `idNumber` and `regimeType` with better types rather than `String`.

```
case class HmrcTaxPeriod(idType: IdType, idNumber: IdNumber, regimeType: RegimeType)
```

Would read much better and would be much more type safe on use site.